describe/describe.go

```diff
@@ -0,0 +1,94 @@
+package describe
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/moov-io/iso8583"
+	"github.com/moov-io/iso8583/field"
+)
+
+func Message(w io.Writer, message *iso8583.Message) error {
+	fmt.Fprintf(w, "ISO 8583 Message:\n")
+
+	printer := fieldPrinter{}
+
+	mti, err := message.GetMTI()
+	if err != nil {
+		return fmt.Errorf("getting MTI: %w", err)
+	}
+	printer.addField("MTI", mti)
+
+	bitmapRaw, err := message.Bitmap().Bytes()
+	if err != nil {
+		return fmt.Errorf("getting bitmap bytes: %w", err)
+	}
+	printer.addField("Bitmap", strings.ToUpper(hex.EncodeToString(bitmapRaw)))
+
+	bitmap, err := message.Bitmap().String()
+	if err != nil {
+		return fmt.Errorf("getting bitmap: %w", err)
+	}
+	printer.addField("Bitmap bits", bitmap)
+
+	// display the rest of all set fields
+	fields := message.GetFields()
+	for _, i := range sortFieldIDs(fields) {
+		// skip the bitmap
+		if i == 1 {
+			continue
+		}
+		field := fields[i]
+		desc := field.Spec().Description
+		str, err := field.String()
+		if err != nil {
+			return fmt.Errorf("getting string value of field %d: %w", i, err)
+		}
+		printer.addField(fmt.Sprintf("F%03d %s", i, desc), str)
+	}
+
+	printer.print()
+
+	return nil
+}
+
+func sortFieldIDs(fields map[int]field.Field) []int {
+	keys := make([]int, 0, len(fields))
+	for k := range fields {
+		keys = append(keys, k)
+	}
+
+	sort.Ints(keys)
+
+	return keys
+}
+
+type printableField struct {
+	description string
+	value       string
+}
+type fieldPrinter struct {
+	fields []printableField
+	maxLen int
+}
+
+func (p *fieldPrinter) addField(description, value string) {
+	field := printableField{description, value}
+	if len(description) > p.maxLen {
+		p.maxLen = len(description)
+	}
+
+	p.fields = append(p.fields, field)
+}
```

Comment:
`text/tabwriter` is super helpful here. You might want to use it.

https://pkg.go.dev/text/tabwriter

Reply:
Thanks @adamdecaf! Initially, I didn't see how I can use it because I didn't want the second column to be aligned. Then I read the docs and examples and found that I just have to omit the trailing tab for the trailing text.

It's more compact now :)